satpy/readers/seviri_base.py

```diff
@@ -910,3 +910,34 @@ def pad_data_vertically(data, final_size, south_bound, north_bound):
     padding_north = get_padding_area(((final_size[0] - north_bound), ncols), data.dtype)
 
     return np.vstack((padding_south, data, padding_north))
+
+
+def mask_bad_quality(data, line_validity, line_geometric_quality, line_radiometric_quality):
+    """Mask scan lines with bad quality.
+
+    For details on quality flags see `MSG Level 1.5 Image Data Format Description`_
+    page 109.
+
+    Args:
+        data (xarray.DataArray):
+            Channel data
+        line_validity (numpy.Array):
+            Quality flags with shape (nlines,).
+        line_geometric_quality (numpy.Array):
+            Quality flags with shape (nlines,).
+        line_radiometric_quality (numpy.Array):
+            Quality flags with shape (nlines,).
+
+    Returns:
+        xarray.DataArray: data with lines flagged as bad converted to np.nan.
+    """
+    # Based on missing (2) or corrupted (3) data
+    print(line_validity)
+    line_mask = line_validity >= 2
+    line_mask &= line_validity <= 3
+    # Do not use (4)
+    line_mask &= line_radiometric_quality == 4
+    line_mask &= line_geometric_quality == 4
+    print(line_mask)
+    data *= np.choose(line_mask, [1, np.nan])[:, np.newaxis].astype(np.float32)
```

Comment:
I think it would be clearer to create a masked copy of the data instead of modifying it inplace. What do you think?

Reply:
Maybe yes. I thought about that too but wanted to keep it as close as possible to the original code for the review.

Reply:
Thank you for thinking of the reviewers ! :smile: 
However, if you see something that can be refactored to be simpler, you don't need to hesitate, just do it!

My suggestion here would be
```suggestion
    line_mask = line_mask[:, np.newaxis]
    data = np.where(line_mask, data, np.nan).astype(np.float32)
```